### PR TITLE
Deprecate various functions related to explicit group encoding

### PR DESCRIPTION
**Pull request by randombit**

### CODE REVIEW

File: src/lib/pubkey/ec_group/ec_group.cpp [hunk 1]
```diff
@@ -706,6 +706,20 @@ EC_Point EC_Group::hash_to_curve(std::string_view hash_fn,
 #endif
 }
 
+std::vector<uint8_t> EC_Group::DER_encode() const {
+   std::vector<uint8_t> output;
+
+   DER_Encoder der(output);
+   const OID oid = get_curve_oid();
+   // TODO(Botan4) this can be removed because an OID will always be defined
```
Comment (reneme) [inline src/lib/pubkey/ec_group/ec_group.cpp:714]: Good idea! I'll pick up this todo markup in the future.

File: src/lib/pubkey/ec_group/ec_group.h [hunk 2]
```diff
@@ -174,10 +174,17 @@ class BOTAN_PUBLIC_API(2, 0) EC_Group final {
...
       * Create the DER encoding of this domain
       * @param form of encoding to use
-      * @returns bytes encododed as DER
+      * @returns the group information encoded as DER
       */
+      BOTAN_DEPRECATED("Use the variant that does not take EC_Group_Encoding")
       std::vector<uint8_t> DER_encode(EC_Group_Encoding form) const;
 
+      /**
+      * Create the DER encoding of this domain, using namedCurve format
```
Comment (reneme) [inline src/lib/pubkey/ec_group/ec_group.h:183]:
```suggestion
      * Create the DER encoding of this domain, using NamedCurve format
```
... I guess.

File: src/lib/pubkey/ecc_key/ecc_key.h [hunk 3]
```diff
@@ -58,7 +58,11 @@ class BOTAN_PUBLIC_API(2, 0) EC_PublicKey : public virtual Public_Key {
       /**
       * Set the domain parameter encoding to be used when encoding this key.
       * @param enc the encoding to use
+      *
+      * This function is deprecated; in a future major release only namedCurve
```
Comment (reneme) [inline src/lib/pubkey/ecc_key/ecc_key.h:62]:
```suggestion
      * This function is deprecated; in a future major release only NamedCurve
```

**Review by reneme — APPROVED** [review]

File: src/lib/pubkey/ec_group/ec_group.h (hunk 2 above)
Comment (randombit) [reply src/lib/pubkey/ec_group/ec_group.h:183]: The RFC calls it `namedCurve`

**Pull request merged**